src/object/obj_rpc.h

```diff
@@ -123,11 +123,13 @@ enum obj_rpc_flags {
 	ORF_RESEND		= (1 << 1),
 	/** Commit DTX synchronously. */
 	ORF_DTX_SYNC		= (1 << 2),
+	/** Reports prior fetch CSUM mismatch */
```

Comment:
I was thinking that client will just report the csum error immediately to control plane through daos agent, any reason we didn't design in this way?
Reporting csum error on next fetch looks not reliable to me, client might do one fetch and hit csum error, then this error won't be logged?

Reply:
The csum error is not discovered on the server, so we have to send it from the client (which discovers the mismatch) to the server.  Liang and I worked this out as the best way to report it to the correct target.  The next fetch just reports the error (calls the log function).  The client then switches from report to retry mode and tries the next replica after this initial follow-up fetch request returns to the client.

I originally planned to to send the error through the control plane (via the agent).  However, the control-plane team didn't like it, and there was a technical issue with reporting to the correct target: there is no way to use cart rank (or any other target identity available in the client) to to route a the csum error to the correct target.

Reply:
The error could be reported directly to the control plane through the agent, but this would require a redo of the health monitor to look elsewhere than the dev_state of the NVMe device for the checksum error counter.

Reply:
then for a non-replica obj csum error (grp_size = 1), it cannot be logged on server forever, right? 
I think better to refine that case.

Reply:
I see, thanks for the explanation. My vision on this kind of monitoring feature is that control plane could be the central repository for all kinds of health information, and it provides notify & query means for the administrator, given this assumption, client should report it's csum error to control plane directly without the NVMe health monitor involved (NVMe health monitor has to be done by io server because the device is exclusively accessed while io server is running), but I agree with you that this design would require much more code changes (especially for control plane).

Probably we may reconsider it once the monitor feature is placed on the agenda.

Reply:
yes, let's reuse the server reporting channel at this point, actually I'm not sure if we should allow client to report csum error to control plane because client cannot determine if the error is ejected by storage or network.

Reply:
Client won't know if the corruption is from network or storage unless we verify checksum on both server and client, that's irrelevant of where should client report the error (report to io server or control plane), right?